### PR TITLE
[FIX] stock: use inventory adjustment name as a reference

### PR DESCRIPTION
**Pull request by DjamelTouati**
**Steps to reproduce the bug:**
- Create a storable product “Test”
- Go to inventory > Operations > Inventory Adjustments
- Add a new line > Choose the “Test” product > Qty = 1*
- Select this line > Click on the “Apply” button
- You have the possibility to enter an “Inventory Reference / Reason”
- Go To reporting > stock moves

**Problem:**
The reference is “Product Quantity Updated” instead of the inventory adjustment name.

In the "action_apply" function, we give the name in the context: https://github.com/odoo/odoo/blob/15.0/addons/stock/wizard/stock_inventory_adjustment_name.py#L27

So that it is then used when creating the `stock.move`: https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_quant.py#L800

But the context is destroyed here because we give a dict instead of a key :  https://github.com/odoo/odoo/blob/15.0/addons/stock/models/stock_quant.py#L296

**Solution:**
Remove the context override because it was done here: https://github.com/odoo/odoo/pull/83884
with the explanation that it's due to the write being called too many times here:
https://github.com/odoo/odoo/blob/8b155b695823dfa954464ad0d5dd07445ece2471/addons/stock/models/stock_quant.py#L244-L248

but then this later PR: https://github.com/odoo/odoo/pull/84771 deletes the same problem lines.

opw-2762169

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_quant.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def action_view_inventory(self):
 
     def action_apply_inventory(self):
         # Update the context to prevent recursive call from write method
-        self = self.with_context({'inventory_report_mode': False})
+        self = self.with_context(inventory_report_mode=False)
```
Comment (ticodoo) [inline addons/stock/models/stock_quant.py:296]: Hello! I started looking into this but didn't finish before you made this PR :smiling_face_with_tear:

I believe we can delete this line (and the comment above it) because it was added here: https://github.com/odoo/odoo/pull/83884 with the explanation that it's due to the write being called too many times here: https://github.com/odoo/odoo/blob/8b155b695823dfa954464ad0d5dd07445ece2471/addons/stock/models/stock_quant.py#L244-L248 but then this later PR: https://github.com/odoo/odoo/pull/84771 deletes the same problem lines.

TLDR: You might be able to delete this line (it doesn't hurt to leave it in, it just might end up being dead code) + a test would be nice too so the feature doesn't break again :smiley:

Comment (DjamelTouati) [reply addons/stock/models/stock_quant.py]: Hello Tiffany
Yes, Indeed, sorry I didn't notice that. I added a test and your explanation in the commit message.
Thank you for your help ;)

**Comment by ticodoo** [comment]
@robodoo r+

Thanks Djamel!

**Comment by ticodoo** [comment]
@robodoo retry
